editor/src/main/java/org/pentaho/pms/ui/util/Splash.java

```diff
@@ -91,12 +91,30 @@ protected Splash( Display display, Shell splashShell, boolean isStartup ) {
     shell.open();
 
     if ( isStartup ) { //show splash for 5s on startup
-      long endTime = System.currentTimeMillis() + 5000;
-      while ( !shell.isDisposed() && endTime > System.currentTimeMillis() ) {
-        if ( isMacOS() && !display.readAndDispatch() ) {
+
+      // Start timer in a separate thread so that it isn't blocked and that it doesn't block clicks to close the
+      // splash early
+      new Thread( () -> {
+        try {
+          Thread.sleep( 5000 ); // Sleep for 5 seconds
```

Comment:
Not a blocker, make this a java constant for `5000` and add comment explaining why 5 seconds.